processor/deltatorateprocessor/processor.go

```diff
@@ -41,10 +49,64 @@ func (dtrp *deltaToRateProcessor) Start(context.Context, component.Host) error {
 
 // processMetrics implements the ProcessMetricsFunc type.
 func (dtrp *deltaToRateProcessor) processMetrics(_ context.Context, md pdata.Metrics) (pdata.Metrics, error) {
+	resourceMetricsSlice := md.ResourceMetrics()
+
+	for i := 0; i < resourceMetricsSlice.Len(); i++ {
+		rm := resourceMetricsSlice.At(i)
+		ilms := rm.InstrumentationLibraryMetrics()
+		for i := 0; i < ilms.Len(); i++ {
+			ilm := ilms.At(i)
+			metricSlice := ilm.Metrics()
+			for j := 0; j < metricSlice.Len(); j++ {
+				metric := metricSlice.At(j)
+				if _, ok := dtrp.ConfiguredMetrics[metric.Name()]; !ok {
+					continue
+				}
+				newDoubleDataPointSlice := pdata.NewNumberDataPointSlice()
+				if metric.DataType() == pdata.MetricDataTypeSum && metric.Sum().AggregationTemporality() == pdata.AggregationTemporalityDelta {
```

Comment:
Probably we should have an error logged if this condition is not happening:

```
if metric.DataType() != pdata.MetricDataTypeSum || metric.Sum().AggregationTemporality() != pdata.AggregationTemporalityDelta {
  logger.Error("Configured metric for rate calculation that is not a sum or delta") // something these lines
  continue
}

```

Reply:
Should this be an error log. I feel like we can print a Debug message. Any thought?

Reply:
Not strong feeling, but probably at least INFO

Reply:
Pushed an update. Thanks.